pycaret/loggers/dagshub_logger.py

```diff
@@ -0,0 +1,94 @@
+import os
+from pathlib import Path
+
+from pycaret.loggers.mlflow_logger import MlflowLogger
+
+try:
+    import dagshub
+    import mlflow
+    from dagshub.upload import Repo
+except ImportError:
+    dagshub = None
+
+
+class DagshubLogger(MlflowLogger):
+    def __init__(self, remote=None) -> None:
+        super().__init__()
+        if dagshub is None:
+            raise ImportError(
+                "DagshubLogger requires dagshub. Install using `pip install dagshub`"
+            )
+        # check token exist or not:
+        token = dagshub.auth.get_token()
+        os.environ["MLFLOW_TRACKING_USERNAME"] = token
+        os.environ["MLFLOW_TRACKING_PASSWORD"] = token
+
+        # Check mlflow environment variable is set:
+        if not remote or "dagshub" not in os.getenv("MLFLOW_TRACKING_URI"):
+            prompt_in = input(
+                "Please insert your repository owner_name/repo_name:"
+            ).split("/")
+            assert (
+                len(prompt_in) == 2
+            ), f"Invalid input, should be owner_name/repo_name, but get {prompt_in} instead"
+
+            dagshub.init(repo_name=prompt_in[1], repo_owner=prompt_in[0])
```

Comment:
what happens if repo already exists?

Reply:
`dagshub.init` manages this. The repository is created only if it does not exist. If it does exist, the environment variables are set up per usual and they can continue to log additional experiments and artifacts to that repository.